cpp/src/arrow/testing/random.h

```diff
@@ -140,6 +140,17 @@ class ARROW_TESTING_EXPORT RandomArrayGenerator {
   std::shared_ptr<Array> Int64(int64_t size, int64_t min, int64_t max,
                                double null_probability = 0);
 
+  /// \brief Generates a random HalfFloatArray
+  ///
+  /// \param[in] size the size of the array to generate
+  /// \param[in] min the lower bound of the uniform distribution
+  /// \param[in] max the upper bound of the uniform distribution
```

Comment:
Hmm... this will not really be a *uniform* float16 distribution if you're selecting random values for the underlying int16 representation, right?

Reply:
Good point. It will not be uniform distribution for float16
Should I only update the comment? Or, should I create uniform distribution float16 random while the current std does not support float16?

Reply:
Updating the comment is enough. As long as the actual values are not processed, it doesn't really matter.